libnmstate/nm/device.py

```diff
@@ -173,62 +173,39 @@ def delete_device(nmdev):
 
 def _safe_delete_device_async(nmdev):
     mainloop = nmclient.mainloop()
-    user_data = mainloop, nmdev
+    user_data = mainloop, nmdev, nmdev.get_iface()
     nmdev.delete_async(
         mainloop.cancellable, _delete_device_callback, user_data
     )
 
 
 def _delete_device_callback(src_object, result, user_data):
-    mainloop, nmdev = user_data
+    mainloop, nmdev, iface = user_data
+    error = None
     try:
-        success = src_object.delete_finish(result)
-    # pylint: disable=catching-non-exception
-    except nmclient.GLib.GError as e:
-        # pylint: enable=catching-non-exception
-        if e.matches(
-            nmclient.Gio.DBusError.quark(),
-            nmclient.Gio.DBusError.UNKNOWN_METHOD,
-        ) or (
-            e.matches(
-                nmclient.NM.DeviceError.quark(),
-                nmclient.NM.DeviceError.NOTSOFTWARE,
-            )
-            and nmdev.is_software()
-        ):
-            logging.debug(
-                "Device %s has been already deleted: error=%s",
-                nmdev.get_iface(),
-                e,
-            )
-            mainloop.execute_next_action()
-        else:
-            mainloop.quit(
-                "Device deletion failed on {}: error={}".format(
-                    nmdev.get_iface(), e
-                )
-            )
-        return
+        src_object.delete_finish(result)
     except Exception as e:
+        error = e
```

Comment:
You may just use `except Exception as error`.

Reply:
Thank you, good catch!

Reply:
Actually, this does not work since it removes `error` from the outer scope, test script:

```python
def test_scope():
    error = "error"
    try:
        raise Exception()
    except Exception as error:
        pass
    print(repr(error))

test_scope()
```

Reply:
Then use another name instead of overriding `error` variable.